lib/hpc/configs.pm

```diff
@@ -16,78 +16,202 @@ use strict;
-sed -i "/^#AccountingStorageHost.*/c\\AccountingStorageHost=$cluster_compute_nodes[-1]" /etc/slurm/slurm.conf
-sed -i "/^#AccountingStorageUser.*/c\\AccountingStoragePort=20088" /etc/slurm/slurm.conf
-sed -i "/^SlurmctldDebug.*/c\\SlurmctldDebug=debug5" /etc/slurm/slurm.conf
-EOF
-        assert_script_run($_) foreach (split /\n/, $config);
-    } else {
-        die "Unsupported value of SLURM_CONF test setting!";
+    $slurm_config{NODES}{NodeName}  = "$cluster_ctl_nodes,$cluster_compute_nodes";
+    $slurm_config{PARTITION}{Nodes} = "$cluster_ctl_nodes,$cluster_compute_nodes";
+    $slurm_config{SlurmctldHost}    = "$cluster_ctl_nodes[0]";
+    $slurm_config{SlurmctldDebug}   = 'debug5';
+
+    if (($slurm_conf eq 'accounting') or ($slurm_conf eq 'nfs_db')) {
+        $slurm_config{JobAcctGatherType}      = 'jobacct_gather/linux';
+        $slurm_config{JobAcctGatherFrequency} = '12';
+        $slurm_config{AccountingStorageType}  = 'accounting_storage/slurmdbd';
+        $slurm_config{AccountingStorageHost}  = "$cluster_compute_nodes[-1]";
+        $slurm_config{AccountingStoragePort}  = '20088';
     }
+    if (($slurm_conf eq 'ha') or ($slurm_conf eq 'nfs_db')) {
+        $slurm_config{SlurmctldHost_tmp} = "$cluster_ctl_nodes[1]";
+        $slurm_config{StateSaveLocation} = '/shared/slurm/';
+        $slurm_config{SlurmctldTimeout}  = '15';
+        $slurm_config{SlurmdTimeout}     = '60';
+    }
+
+    script_run("echo '#slurm.conf generated by the tests' > /etc/slurm/slurm.conf");
+    my @pairs = ();
+    my $NODES = '';
+    my $PARTITION = '';
+    while ( my ($key,$value) = each %slurm_config) {
+        push(@pairs, $key . '=' . $value) unless ($key eq 'NODES') or ($key eq 'PARTITION')
+            or ($key eq 'SlurmctldHost_tmp');
+        if ($key eq 'SlurmctldHost_tmp') {
+            my $tmp_key = 'SlurmctldHost';
+            push(@pairs, $tmp_key . '=' . $value);
+        }
+        if ($key eq 'NODES') {
+            my @pairs = ();
+            while ( my ($key,$value) = each %{$slurm_config{NODES}}) {
+                push(@pairs, $key . '=' . $value);
+            }
+            $NODES = join(' ', @pairs);
+        }
+        if ($key eq 'PARTITION') {
+            my @pairs = ();
+            while ( my ($key,$value) = each %{$slurm_config{PARTITION}}) {
+                push(@pairs, $key . '=' . $value);
+            }
+            $PARTITION = join(' ', @pairs);
+        }
+    }
+    my $slurm_config_tmp = join('+', @pairs);
+    script_run("echo $slurm_config_tmp | tee -a /etc/slurm/slurm.conf");
+    script_run("echo $NODES | tee -a /etc/slurm/slurm.conf");
+    script_run("echo $PARTITION | tee -a /etc/slurm/slurm.conf");
+    script_run("tr '+' '\\n' < /etc/slurm/slurm.conf > /etc/slurm/slurm.conf_tmp");
+    script_run("mv /etc/slurm/slurm.conf_tmp /etc/slurm/slurm.conf");
 }
 
+=head2
+
+Prepare slurmdbd.conf based on test requirements and settings
```

Comment:
typo I think? seems like one 'd' to much in the filename

Reply:
yeah, as we talked in person; this is slurm database daemon which indeed is called slurmdbd